html/mycakeapp/src/Controller/AuctionController.php

```diff
@@ -83,17 +87,38 @@ public function view($id = null)
 	public function add()
 	{
 		// Biditemインスタンスを用意
+		// 挿入
 		$biditem = $this->Biditems->newEntity();
 		// POST送信時の処理
 		if ($this->request->is('post')) {
-			// $biditemにフォームの送信内容を反映
-			$biditem = $this->Biditems->patchEntity($biditem, $this->request->getData());
-			// $biditemを保存する
+			// データ取得
+			$data = $this->request->getData();
+			// データ挿入
+			$biditem = $this->Biditems->newEntity($data);
+			$biditem->image_name = $_FILES['image_name']['name'];
+			// id,image_name以外を一旦保存			
+			$this->Biditems->save($biditem);
+			// 更新
+			$biditem = $this->Biditems->patchEntity($biditem, $data);
+			// pathinfoで配列で拡張子取り出す
+			$path_parts = pathinfo($biditem['image_name']);
+			// 拡張子を変数に入れる
+			$fileExt = $path_parts["extension"];
```

Comment:
```suggestion
			$fileExt = $path_parts["extension"] ?? '';
```
現状「dummyjpeg」のようにピリオド.無しのファイルを送るとここのパス解析でワーニングが出るのでこうしましょう

Reply:
> 現状「dummyjpeg」のようにピリオド.無しのファイルを送るとここのパス解析でワーニングが出るのでこうしましょう

```php
$fileExt = $path_parts["extension"] ?? '';
```
が先祖返りしてしまっているので元に戻しましょう。ファイル名「dummyjpeg」を送るとワーニングが出ます